clang/lib/DPCT/ExtReplacements.cpp

```diff
@@ -325,8 +325,10 @@ void ExtReplacements::addReplacement(std::shared_ptr<ExtReplacement> Repl) {
     ReplMap.insert(std::make_pair(Repl->getOffset(), Repl));
     // If Repl is insert replacement, check whether it is alive or dead.
   } else if (checkLiveness(Repl)) {
-    if (Repl->IsSYCLHeaderNeeded())
-      FileInfo->insertHeader(HT_SYCL);
+    if (Repl->getReplacementText().find("#include") == std::string::npos) {
```

Comment:
There is risk to check the content of the replacement.   (eg.  memcpy(dst,  "#inlcude", size).  
So please redesin the logic like introduceing extra flag for repleacement.